### PR TITLE
fix issue 15818 - Multiple function declarations without definition cause ambiguity overloading error

### PR DESCRIPTION
**Pull request by ghost**
The compiler will assume that all identical body-less functions will be linked to the same stuff.

### CODE REVIEW

File: src/dmd/dtemplate.d [hunk 1]
```diff
@@ -2543,6 +2546,7 @@ void functionResolve(ref MatchAccumulator m, Dsymbol dstart, Loc loc, Scope* sc,
             {
                 if (fd.fbody && !m.lastf.fbody) goto LfIsBetter;
                 if (!fd.fbody && m.lastf.fbody) goto LlastIsBetter;
+                if (!fd.fbody && !m.lastf.fbody) goto LlastIsBetter;
```
Comment (dkorpel) [inline src/dmd/dtemplate.d]: Can't you just have one `if (!fd.fbody) goto LlastIsBetter;` ?

Comment (Basile-z) [reply src/dmd/dtemplate.d]: Yeah but maybe the current way it's written is more clear, so I won't gonna change it. Is it okay as justification ?

Comment (dkorpel) [reply src/dmd/dtemplate.d]: When I read it I think "why is there essentially a tautology `(!x || x)` there, is the goto label slightly different? Is fbody a function with a side-effect so it's not a tautology after all?". So I don't agree it's clearer. OTOH I don't want to bikeshed, and ultimately it's up to someone with merge rights to approve it.

Comment (Basile-z) [reply src/dmd/dtemplate.d]: the semantic check is clear when the the relational expression is an andand.

Comment (RazvanN7) [reply src/dmd/dtemplate.d]: What happens when both have a body? If we apply your reasoning to the end, we should express that case also (with an empty if body).

Comment (Basile-z) [reply src/dmd/dtemplate.d]: ok done.

**Review by thewilsonator — APPROVED** [review]

**Pull request merged**